test/datagenerator/1800medicare.go

```diff
@@ -0,0 +1,161 @@
+		}
+
+		s := bufio.NewScanner(inf)
+
+		for s.Scan() {
+			// Randomly select HICNs from ACO for suppression records
+			if rand.Intn(2) == 0 {
+				continue
+			}
+
+			hicn := string(s.Bytes())
+			p := profile(hicn, acoID)
+			recs := records(p)
+			for _, r := range recs {
+				for _, f := range []field{r.hicn, r.blk, r.firstName, r.middleName, r.lastName, r.dob, r.addr1, r.addr2, r.addr3, r.city, r.state, r.zip5, r.zip4, r.gender, r.encDate, r.effDate, r.srcCode, r.mechCode, r.prefInd, r.saEffDate, r.saSrcCode, r.saMechCode, r.saPrefInd, r.acoID, r.acoName} {
+					_, err = outf.WriteString(fmt.Sprintf("%-"+fmt.Sprint(f.l)+"s", f.v))
+					if err != nil {
+						panic(err)
+					}
+				}
+				_, err = outf.WriteString("\n")
+				if err != nil {
+					panic(err)
+				}
+			}
+		}
+
+		err = inf.Close()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	err = outf.Close()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func profile(hicn string, acoID string) record {
+	dobMin, _ := time.Parse("2006-01-02", "1900-01-01")
+	dobMax := time.Now().Add(-65 * 365 * 24 * time.Hour)
+
+	p := record{
+		hicn:       field{l: 11, v: hicn},                     // HICN
+		blk:        field{l: 10},                              // Beneficiary link key
+		firstName:  field{l: 30, v: randWord(1, 30)},          // Beneficiary first name
+		middleName: field{l: 30},                              // Beneficiary middle name
+		lastName:   field{l: 40, v: randWord(1, 30)},          // Beneficiary last name
+		dob:        field{l: 8, v: ccyymmdd(dobMin, dobMax)},  // Beneficiary date of birth
+		addr1:      field{l: 55, v: addr1()},                  // Beneficiary address line 1
+		addr2:      field{l: 55},                              // Beneficiary address line 2
+		addr3:      field{l: 55},                              // Beneficiary address line 3
+		city:       field{l: 40, v: randWord(1, 40)},          // Beneficiary city
+		state:      field{l: 2, v: "ST"},                      // Beneficiary state
+		zip5:       field{l: 5, v: "00000"},                   // Beneficiary first five digits of ZIP code
+		zip4:       field{l: 4},                               // Beneficiary last four digits of ZIP code
+		gender:     field{l: 1, v: oneOfStr("M", "F", "U")},   // Beneficiary gender
+		acoID:      field{l: 5, v: acoID},                     // ACO identifier
+		acoName:    field{l: 70, v: randWord(1, 66) + " ACO"}, // ACO legal name
+	}
+
+	return p
+}
+
+func records(profile record) []record {
+	// Create 0-5 suppression records for this profile
+	ct := rand.Intn(6)
+	encDtMin, encDtMax := time.Now().Add(-7*24*time.Hour), time.Now()
+	effDtMin, effDtMax := time.Now().Add(-7*24*time.Hour), time.Now().Add(3*24*time.Hour)
+	records := []record{}
+
+	for i := 0; i < ct; i++ {
+		r := profile
+		r.encDate = field{l: 8, v: ccyymmdd(encDtMin, encDtMax)} // Encounter date
+		r.effDate = field{l: 8, v: ccyymmdd(effDtMin, effDtMax)} // Beneficiary data sharing effective date
```

Comment:
It'd be cool if we could throw a blank value for the effective date, just like the other fields since that's a valid option. It'd make our testing even more robust!